### PR TITLE
Routing kit preprocessing # ------------------------ >8 ------------------------ Do not modify or remove the line above.  # Everything below it will be ignored.

### PR DESCRIPTION
**Pull request by r-barnes**
Requesting a pull to nickolasclarke:master from nickolasclarke:richard/rk_preprocess

Write a message for this pull request. The first block
of text is the title and the rest is the description.

Changes:

28449c1 (Richard Barnes, 47 seconds ago)
   Add margins

0c3fbef (Richard Barnes, 2 hours ago)
   Make a note in example usage to get highways file

5243394 (Richard Barnes, 2 hours ago)
   Standardize data storage locations and detect nans in extents

d164d5a (Richard Barnes, 2 hours ago)
   Add progress to osm_splitter and clean up variabel names and input

8353f76 (Richard Barnes, 2 hours ago)
   Add more usage examples

0f17f7c (Richard Barnes, 23 hours ago)
   Improve extents error handling

b1264a6 (Richard Barnes, 23 hours ago)
   Drop bboxSplit.cpp

fe1f03b (Richard Barnes, 23 hours ago)
   Add an 'extents' command to pull_gtfs.py

74637f9 (Richard Barnes, 2 days ago)
   OSM Splitter seems to be working

4061e5e (Richard Barnes, 2 days ago)
   Upgrades to bboxSplit.cpp to fix warnings

cccf853 (Richard Barnes, 2 days ago)
   Initial import of bboxSplit.cpp

f14d18a (Richard Barnes, 3 days ago)
   Add file sizes

3009b07 (Richard Barnes, 3 days ago)
   Add timing results for contraction hierarchy

faac731 (Richard Barnes, 5 days ago)
   Add processing times for osmosis

08e9085 (Richard Barnes, 5 days ago)
   Add instructions for extraction

1112a70 (Richard Barnes, 5 days ago)
   Added preprocessing for routing

8544cbe (Richard Barnes, 5 days ago)
   Downgrade cmake required version

27d1d85 (Richard Barnes, 5 days ago)
   Added preprocessing for routing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,11 +31,50 @@ Now build with the following
...
+
+#Get OSM extents (caches for faster runs a second time)
+./build/bin/pull_gtfs.py extents data/feeds.db > data/extents
+
+#Get a Planet OSM highways file per the instructions below
+
+#Split up global highway data padding each box with a 2 mile margin
+./build/bin/osm_splitter data/planet-highways.osm.pbf data/extents 0.0288
+
+#Run model on a dataset
```
Comment (nickolasclarke) [inline README.md:70]: Is this out of execution order? You should parse gtfs first

File: src/parse_gtfs.py [hunk 2]
```diff
@@ -377,6 +377,19 @@ def HasBlockIDs(gtfs_filename):
 
 
 
+def GetExtents(gtfs_filename):
+  """Returns a bounding box containing all of the feed's stops.
+
+  Returns: [minlon, minlat, maxlon, maxlat]
+  """
+  gtfs   = ptg.load_geo_feed(gtfs_filename)
+  bounds = gtfs.stops.total_bounds
```
Comment (nickolasclarke) [inline src/parse_gtfs.py:386]: That's handy!

**Review by nickolasclarke — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (r-barnes) [reply README.md:70]: Fixed in current version.

File: src/parse_gtfs.py (hunk 2 above)
Comment (r-barnes) [reply src/parse_gtfs.py:386]: Yeah, though it's since required a little cleanup for validation.

**Pull request merged**